src/plugins/git/createGraph.js

```diff
@@ -0,0 +1,132 @@
+// @flow
+
+import type {Address} from "../../core/address";
+import type {Node, Edge} from "../../core/graph";
+import type {
+  Repository,
+  Commit,
+  Tree,
+  TreeEntry,
+  NodePayload,
+  EdgePayload,
+  NodeType,
+  EdgeType,
+} from "./types";
+import {Graph, edgeID} from "../../core/graph";
+import {GIT_PLUGIN_NAME, treeEntryId} from "./types";
+
+class GitGraphCreator {
+  repositoryName: string;
+
+  constructor(repositoryName) {
+    this.repositoryName = repositoryName;
+  }
+
+  makeAddress(type: NodeType | EdgeType, id: string): Address {
+    return {
+      pluginName: GIT_PLUGIN_NAME,
+      repositoryName: this.repositoryName,
+      type,
+      id,
+    };
+  }
+
+  createGraph(repository: Repository): Graph<NodePayload, EdgePayload> {
+    const graphs = [
+      ...Object.keys(repository.commits).map((hash) =>
+        this.commitGraph(repository.commits[hash])
+      ),
+      ...Object.keys(repository.trees).map((hash) =>
+        this.treeGraph(repository.trees[hash])
+      ),
+    ];
+    return graphs.reduce((g, h) => Graph.mergeConservative(g, h), new Graph());
+  }
+
+  commitGraph(commit: Commit) {
+    const commitNode = {
+      address: this.makeAddress("COMMIT", commit.hash),
+      payload: {},
+    };
+    const treeNode = {
+      address: this.makeAddress("TREE", commit.treeHash),
+      payload: {},
+    };
+    const edge = {
+      address: this.makeAddress(
+        "HAS_TREE",
+        edgeID(commitNode.address, treeNode.address)
+      ),
+      src: commitNode.address,
+      dst: treeNode.address,
+      payload: {},
+    };
+    return new Graph()
+      .addNode(commitNode)
+      .addNode(treeNode)
+      .addEdge(edge);
+  }
+
+  treeGraph(tree: Tree) {
+    const treeNode = {
+      address: this.makeAddress("TREE", tree.hash),
+      payload: {},
+    };
+    const result = new Graph().addNode(treeNode);
+    Object.keys(tree.entries).forEach((name) => {
+      const entry = tree.entries[name];
+      const entryNode = {
+        address: this.makeAddress(
+          "TREE_ENTRY",
```

Comment:
Here and elsewhere: use the variable rather than string constant

Reply:
Done.